src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -237,14 +238,17 @@ def _try_specializing_pnad(
         necessary_add_effects: Set[GroundAtom],
         pnad: PartialNSRTAndDatastore,
         segment: Segment,
+        check_datastore_change: bool = True
     ) -> Optional[PartialNSRTAndDatastore]:
         """Given a PNAD and some necessary add effects that the PNAD must
         achieve, try to make the PNAD's add effects more specific
         ("specialize") so that they cover these necessary add effects.
 
         Returns the new constructed PNAD, without modifying the
         original. If the PNAD does not have a grounding that can even
-        partially satisfy the necessary add effects, returns None.
+        partially satisfy the necessary add effects, or if the newly
```

Comment:
Fix this docstring -- this second condition only matters if `check_datastore_change` is True.